### PR TITLE
Texlive 2023

### PR DESCRIPTION
**Pull request by fosslinux**
#### Testing the changes
- I tested the changes in this PR: **yes**

#### Local build testing
- I built this PR locally for my native architecture, x86_64
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [x] aarch64-musl
  - [x] armv7l
  - [x] armv6l-musl

<strike>A few consideration's I'm 100% sure about.

1. I'm not certain about our current ownerships - a lot of things from `texmf-dist` come from the texlive package, whereas most other distributions appear to have the primary texlive package as compiled binaries-only and then place all the `texmf-dist` binaries symlinked to scripts in their texlive-core package (obviously different names but same idea). Should we be doing this?
2. What do I do about the ppc64 patches, should they be dropped or left?</strike>
*all considered now*

### CODE REVIEW

**Comment by tornaria** [comment]
Small suggestion:
```
--- a/common/build-style/texmf.sh
+++ b/common/build-style/texmf.sh
@@ -3,14 +3,11 @@ do_build() {
        # Extract the source files
        mkdir -p "build/usr/share/texmf-dist"
        find . -maxdepth 1 -print -name "*.tar.xz" \
-               -exec bsdtar -C "build/usr/share/texmf-dist" -xf {} \;
+               -exec bsdtar \
+                       -s '|^texmf-dist/||' \
+                       -C "build/usr/share/texmf-dist" \
+                       -xf {} \;
        cd "build/usr/share/texmf-dist/"
-       # Everything in usr/share/texmf-dist/texmf-dist should really be in
-       # usr/share/texmf-dist, so we move it
-       if [ -d "texmf-dist" ] ; then
-               rsync -ar texmf-dist/ ./
-               rm -rf texmf-dist/
-       fi
        # LICENSEs are unneeded
        rm -f LICENSE*
 
```

If you do this, maybe get rid of the rsync makedepends ?


Do you think there's a chance we can try to fix `texdoc` ? (#30340)

**Comment by tornaria** [comment]
Also note this:
```diff
--- a/srcpkgs/texlive/patches/tlmgr.patch
+++ b/srcpkgs/texlive/patches/tlmgr.patch
@@ -63,7 +63,7 @@
 +  if (!$opts{"usermode"} && $action ne "init-usertree") {
 +    $opts{"usermode"} = 1;
 +    print "(running on Void Linux, switching to user mode!)\n";
-+    print "(see https://docs.voidlinux.org/config/texlive.html)\n");
++    print "(see https://docs.voidlinux.org/config/texlive.html)\n";
 +  }
 +
    #
```

**Comment by fosslinux** [comment]
Thanks; applied those two.

Re: texdoc, I'm not certain of the best way to attack that at this point. We can't include the docs in the main package - they're HUGE. They don't seem to be included in any tarball. This is something I'd prefer to address in a follow up PR.

**Comment by tornaria** [comment]
> Thanks; applied those two.
> 
> Re: texdoc, I'm not certain of the best way to attack that at this point. We can't include the docs in the main package - they're HUGE. They don't seem to be included in any tarball. This is something I'd prefer to address in a follow up PR.

I think the docs are (optionally?) produced when each TL package is built, and placed in `$TEXMFDIST/doc/`. I think arch creates these TL packages without the documentation (each `*.xz` file inside `texlive-core-2023.66587-src.zip` and friends is one TL package) and makes this zip files with collections of packages.

In spite of the name, I don't think these files `texlive-core-2023.66587-src.zip` are texlive sources, and I don't think the arch package uses those, rather something more like `texlive-20230313-texmf.tar.xz`. That one file seems to contain everything, source, runtime and documentation for all the collections together. Then there's `texlive-20230311-tlpdb-full.tar.gz` which contains the database of installed files; arch seems to parse the database to split the texmf runtime files into collections, and I think all the documentation goes together in a single package (but documentation could also be split by collections).

TL;DR
 - the docs are included in `texlive-20230313-texmf.tar.xz`
 - the database with meta information of all those files is in `texlive-20230311-tlpdb-full.tar.gz`
 - maybe it's possible to eventually transition to using those files to generate all collections and documentation of collections (and also ship the database which I think is needed by texdoc)


Another note: I think texlive expects the `tlpkg` directory to be placed in `/usr/share/tlpkg` instead of `/usr/share/texmf-dist/tlpkg`. Simple way to fix this is to make a symlink, and I think doing this gets rid of some patches here.

**Comment by fosslinux** [comment]
Hmm, alright.

I'll have a bit of a closer look at arch's texlive packages, and a couple of other distributions.
If it is appropriate to do so, I'll have a look at the possibility of taking collections straight from upstream, rather than from Arch.

Previously, the consensus was that shipping tlpdb didn't really make any sense since tlmgr system wide is unsupported. But that might not be quite true. Perhaps tlpdb should be shipped - with the intention that it is still a package managed file and is not changed by the user.

(This is still something I'd prefer to address after this PR, let's get this merged first)

**Comment by return42** [comment]
@fosslinux TeX-Live 2025 has [been released](https://www.ctan.org/tex-archive/systems/texlive/Source) .. if you have time for, could you prepare a new PR to upgrade Tex-Live?
